sound/soc/sof/intel/hda.c

```diff
@@ -992,9 +992,15 @@ static int hda_generic_machine_select(struct snd_sof_dev *sdev)
 				dmic_num = hda_dmic_num;
 
 			switch (dmic_num) {
+			case 1:
+				dmic_str = "-1ch";
+				break;
 			case 2:
 				dmic_str = "-2ch";
 				break;
+			case 3:
+				dmic_str = "-3ch";
+				break;
 			case 4:
 				dmic_str = "-4ch";
 				break;
```

Comment:
you could do an `snprintf()` already, or even a simple assignment like
```
char dmic_str[] = "-2ch";
dmic_str[1] = '0' + dmic_num;
```
but that might be too ugly ;-)

Reply:
yeah was just trying to get away with as small change as possible, but you're right if we for e.g. start to have dmics up to 8 this doesn't look really nice anymore :)

Reply:
agree with minimal changes here, it's not clear if this is a one-off or a real trend.

Reply:
yes closing this now as talking to @lyakh, let's optimize in later PR's if needed